web/groups.go

```diff
@@ -193,11 +204,11 @@ func (s *AutograderService) getGroupUsers(request *pb.Group) ([]*pb.User, error)
 }
 
 // isValidGroupName ensures that SCM team and repository names for the given group
-// will not coincide with one of the existing groups
+// will not coincide with one of the existing approved groups
 func (s *AutograderService) isValidGroupName(courseID uint64, groupName string) bool {
 	courseGroups, _ := s.db.GetGroupsByCourse(courseID)
 	for _, group := range courseGroups {
-		if slug.Make(groupName) == slug.Make(group.GetName()) {
+		if slug.Make(groupName) == slug.Make(group.GetName()) && group.GetStatus() == pb.Group_APPROVED {
```

Comment:
Suggest to change `db.GetGroupsByCourse` to take an optional query parameter that can reduce the number of returned groups to those with a given group status (e.g. `pb.Group_PENDING` or `pb.Group_APPROVED`)? The signature would look something like: 
```go
	GetGroupsByCourse(cid uint64, statuses ...pb.Group_GroupStatus) ([]*pb.Group, error)
```
This change can be implemented without changing existing uses of `GetGroupsByCourse()`, but maybe other uses can take advantage of the extra query parameter. This would be similar to `GetEnrollmentsByCourse` and `GetCoursesByUser`, which already support a similar style.

With such a change, you would avoid this check that the group is approved, while looping over them.

Reply:
Sure, but I better add this to the gorm related PR #203 to avoid unnecessary conflicts.

Reply:
Now that #203 has been merged, we should change the above before I merge this PR.

Reply:
Fixed now.